### PR TITLE
Add example for NDT sensor model in beluga_example

### LINKED ISSUE #350: Create NDT AMCL node example launchfile in `beluga_example`
### Feature description

Exactly as the title says. Using the turtlebot map we already have checked into the repo.
This includes providing sensible defaults for the parameters.

### PR DESCRIPTION
**Pull request by serraramiro1**
### Proposed changes

Adds a new launch file in `beluga_example` for using the new `NDT` sensor model, providing sensible defaults.
Also, modifies the likelihood computation of the sensor model to use a 3x3 cells neighborhood instead of just 1x1 that's very susceptible to boundary effects.

https://github.com/Ekumen-OS/beluga/assets/53626539/1f2be054-7c8b-4e19-99f0-6283dc6a299e

Closes #350 

#### Type of change

- [ ] 🐛 Bugfix (change which fixes an issue)
- [x] 🚀 Feature (change which adds functionality)
- [ ] 📚 Documentation (change which fixes or extends documentation)

### Checklist

_Put an `x` in the boxes that apply. This is simply a reminder of what we will require before merging your code._

- [x] Lint and unit tests (if any) pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] All commits have been signed for [DCO](https://developercertificate.org/)

### CODE REVIEW

File: beluga_example/launch/utils/ndt_localization_launch.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+        DeclareLaunchArgument(
+            name='localization_map',
+            default_value=str(example_dir_path / 'maps' / 'turtlebot3_world.yaml'),
+            description='Map YAML file to be used by the map server node.',
+        ),
+        DeclareLaunchArgument(
+            name='ndt_map',
+            default_value=str(example_dir_path / 'maps' / 'turtlebot3_world.hdf5'),
+            description='Map HDF5 file used by the localization node.',
+        ),
```
Comment (hidmic) [inline beluga_example/launch/utils/ndt_localization_launch.py:62]: @serraramiro1 meta: do we need to pass the `ndt_map` explicitly? `get_node_with_arguments_declared_from_params_file` already declares arguments based on ROS 2 parameters YAML file contents.

File: beluga/include/beluga/sensor/ndt_sensor_model.hpp [hunk 2]
```diff
@@ -165,14 +168,23 @@ class NDTSensorModel {
...
+  /// the 3x3 cells around the measurement cell, or 'params_.min_likelihood', whichever is higher.
   [[nodiscard]] double likelihood_at(const ndt_cell_type& measurement) const {
-    const auto maybe_cell = cells_data_.data_near(measurement.mean);
-    if (!maybe_cell.has_value()) {
-      return params_.minimum_likelihood;
+    static const std::array kCellsToConsider{
+        Eigen::Vector2i{-1, -1}, Eigen::Vector2i{-1, -0}, Eigen::Vector2i{-1, 1},
+        Eigen::Vector2i{0, -1},  Eigen::Vector2i{0, 0},   Eigen::Vector2i{0, 1},
+        Eigen::Vector2i{1, -1},  Eigen::Vector2i{1, 0},   Eigen::Vector2i{1, 1},
+    };
```
Comment (hidmic) [inline beluga/include/beluga/sensor/ndt_sensor_model.hpp:178]: @serraramiro1 meta: some day it would be nice for this kernel to be configurable.

**Review by hidmic — COMMENTED** [review]
First pass. Haven't tested it locally. Localization in video looks a bit wiggly. We are on track though.

File: beluga/include/beluga/sensor/ndt_sensor_model.hpp (hunk 2 above)
Comment (serraramiro1) [reply beluga/include/beluga/sensor/ndt_sensor_model.hpp]: Parametrized it but didn't add the configuration point for the ros node as it didn't seem worth it.
We can always add that later.

File: beluga_example/launch/utils/ndt_localization_launch.py (hunk 1 above)
Comment (serraramiro1) [reply beluga_example/launch/utils/ndt_localization_launch.py:62]: Unfortunately this way of passing parameters doesn't support variable substituion, that's why I went for this approach.
Happy to use another

Comment (hidmic) [reply beluga_example/launch/utils/ndt_localization_launch.py:62]: That's a good point. That magic is incomplete. Oh well, nevermind.

File: beluga_example/launch/utils/ndt_localization_launch.py [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+            default_value=str(example_dir_path / 'params' / 'default_ndt.ros2.yaml'),
+            description='Parameters file to be used to run the localization node.',
+        ),
+        DeclareLaunchArgument(
+            name='localization_map',
+            default_value=str(example_dir_path / 'maps' / 'turtlebot3_world.yaml'),
+            description='Map YAML file to be used by the map server node.',
+        ),
+        DeclareLaunchArgument(
+            name='ndt_map',
```
Comment (hidmic) [inline beluga_example/launch/utils/ndt_localization_launch.py:59]: @serraramiro1 nit^2: consider `localization_ndt_map` for consistency.

File: beluga_example/launch/utils/ndt_localization_launch.py [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+    use_composition,
+    use_sim_time,
+):
+    use_composition = get_typed_value(use_composition, bool)
+    use_sim_time = get_typed_value(use_sim_time, bool)
+
+    load_nodes = GroupAction(
+        actions=[
+            *get_node_with_arguments_declared_from_params_file(
+                package="beluga_amcl",
```
Comment (hidmic) [inline beluga_example/launch/utils/ndt_localization_launch.py:92]: @serraramiro1 nit: use single quotes everywhere for consistency.

File: beluga_example/rviz/ndt_rviz.ros2.rviz [hunk 5]
```diff
@@ -0,0 +1,289 @@
+Panels:
```
Comment (hidmic) [inline beluga_example/rviz/ndt_rviz.ros2.rviz:1]: @serraramiro1 nit^N!: maybe it's about time we rename rviz config files to match their use: `ndt_amcl.ros2.rviz` here, `amcl.ros*.rviz` elsewhere.

**Review by hidmic — APPROVED** [review]

**Comment by serraramiro1** [comment]
@hidmic Addressed comments, mind taking a last look?

**Pull request merged**